File: data/compilers_comparation/src/examples/second_example/src/func39.c
#include "second_example.h" 
array_t* func39(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array112;
   if (pCounter > 0) {
      array112 = vars->data[--pCounter];
      array112->refC++;
      DEBUG_COPY(array112->id);
   } else {
      array112 = (array_t*)malloc(sizeof(array_t));
      array112->size = 994;
      array112->refC = 1;
      array112->id = 112;
      array112->data = (unsigned int*)malloc(array112->size*sizeof(unsigned int));
      memset(array112->data, 0, array112->size*sizeof(unsigned int));
      DEBUG_NEW(array112->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array112;
   array_t* array113 = func41(&params0, loopsFactor);
   DEBUG_RETURN(array113->id);
   free(params0.data);
   if(PATH0 & 1) {
      array_t* array114;
      if (pCounter > 0) {
         array114 = vars->data[--pCounter];
         array114->refC++;
         DEBUG_COPY(array114->id);
      } else {
         array114 = (array_t*)malloc(sizeof(array_t));
         array114->size = 769;
         array114->refC = 1;
         array114->id = 114;
         array114->data = (unsigned int*)malloc(array114->size*sizeof(unsigned int));
         memset(array114->data, 0, array114->size*sizeof(unsigned int));
         DEBUG_NEW(array114->id);
      }
      unsigned int loop35 = 0;
      unsigned int loopLimit35 = (rand()%loopsFactor)/4 + 1;
      for(; loop35 < loopLimit35; loop35++) {
         for (int i = 0; i < array112->size; i++) {
            array112->data[i]--;
         }
         for (int i = 0; i < array114->size; i++) {
            if (array114->data[i] == 50) { 
               return array114;
            }
         }
      }
      array_t_param params1;
      params1.size = 3;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array112;
      params1.data[1] = array113;
      params1.data[2] = array114;
      array_t* array115 = func54(&params1, rng(), loopsFactor);
      DEBUG_RETURN(array115->id);
      free(params1.data);
      array115->refC--;
      if(array115->refC == 0) {
         free(array115->data);
         free(array115);
         DEBUG_FREE(array115->id);
      }
      array114->refC--;
      if(array114->refC == 0) {
         free(array114->data);
         free(array114);
         DEBUG_FREE(array114->id);
      }
   }
   else {
      array_t_param params1;
      params1.size = 2;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array112;
      params1.data[1] = array113;
      array_t* array116 = func50(&params1, loopsFactor);
      DEBUG_RETURN(array116->id);
      free(params1.data);
      array116->refC--;
      if(array116->refC == 0) {
         free(array116->data);
         free(array116);
         DEBUG_FREE(array116->id);
      }
   }
   array113->refC--;
   if(array113->refC == 0) {
      free(array113->data);
      free(array113);
      DEBUG_FREE(array113->id);
   }
   return array112;
}

File: data/multilang_comparison/program_go/program_go/src/func39.go
package main
import "math/rand"
var _ = rand.Intn
func Func39(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array82 *Array
   if pCounter > 0 {
       array82 = vars.Data[pCounter-1]
       pCounter--
       array82.RefC++
   } else {
       array82 = &Array{
           Size: 71,
           RefC: 1,
           Id: 82,
           Data: make([]uint32, 71),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array82
   var array83 *Array = Func41(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   if (PATH0 & 1) != 0 {
      var array84 *Array
      if pCounter > 0 {
          array84 = vars.Data[pCounter-1]
          pCounter--
          array84.RefC++
      } else {
          array84 = &Array{
              Size: 881,
              RefC: 1,
              Id: 84,
              Data: make([]uint32, 881),
          }
      }
      var loop26 int = 0;
      var loopLimit26 int = (50)/4 + 1;
      for ; loop26 < loopLimit26; loop26++ {
         for i := 0; i < array83.Size; i++ {
             array83.Data[i]--
         }
         for i := 0; i < array83.Size; i++ {
             if array83.Data[i] == 94 {
                 return array83
             }
         }
      }
      array84.RefC--
      if array84.RefC == 0 {
          array84.Data = nil
          array84 = nil
      }
   
   } else {
      var params1 ArrayParam
      params1.Size = 2
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array82
      params1.Data[1] = array83
      var array85 *Array = Func50(&params1, loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array85.RefC--
      if array85.RefC == 0 {
          array85.Data = nil
          array85 = nil
      }
   }
   array82.RefC--
   if array82.RefC == 0 {
       array82.Data = nil
       array82 = nil
   }
   return array83;
}

